### PR TITLE
DAOS-12888 test: skip non data device checking

### PR DESCRIPTION
**Pull request by wangshilong**
Test-nvme: auto_md_on_ssd
Skip-checkpatch: true
Quick-Functional: true
Skip-unit-tests: true
Skip-nlt: true
Test-tag: test_csum_error_logging
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by wangshilong** [comment]
Test passed: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12011/16/artifact/Functional%20Hardware%20Medium/launch/functional_hardware_medium/38-checksum-csum_error_logging-launch.log/*view*/

File: src/tests/ftest/checksum/csum_error_logging.py [hunk 1]
```diff
@@ -64,11 +64,17 @@ def test_csum_error_logging(self):
                 if 'tgt_ids' not in device or 'uuid' not in device:
                     self.fail(
                         'Missing uuid and/or tgt_ids info from dmg storage query list devices')
+                if 'role_bits' not in device:
+                    self.fail(
+                        'Missing role bits info from dmg storage query list devices')
```
Comment (NiuYawei) [inline src/tests/ftest/checksum/csum_error_logging.py:69]: Will this work for PMEM mode?

Comment (wangshilong) [reply src/tests/ftest/checksum/csum_error_logging.py:69]: I checked this with PMEM mode, output is same, so it should work.
--------
wolf-150
--------
  Devices
    UUID:f8edbb4f-21c9-4ce9-b7d3-79481fefb511 [TrAddr:0000:67:00.0]
      Roles:data Targets:[0] Rank:0 State:NORMAL LED:NA

**Review by NiuYawei — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Comment by wangshilong** [comment]
> Test passed: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12011/16/artifact/Functional%20Hardware%20Medium/launch/functional_hardware_medium/38-checksum-csum_error_logging-launch.log/*view*/

@daos-stack/daos-gatekeeper would you force to land this PR, PR only modified csum_error test, and it passed. other failures  are not related to this PR.

**Pull request merged**